### PR TITLE
Anf 7859 2020 06 01 api bugfix snapshot policy list volume

### PR DESCRIPTION
**Pull request by audunn**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

This is and urgent fix that addresses a bug that was discovered in generated SDK's. The generated code is unable to reserialize snapshot policy list volume response (the spec did not match the response) and needs to be addressed in 2020-06-01 and 2020-07-01.

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [x] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by audunn** [comment]
@lirenhe  can we get this moving quick?  The breaking change is to address a broken functionality for customers/users.

**Comment by lirenhe** [comment]
> @lirenhe can we get this moving quick? The breaking change is to address a broken functionality for customers/users.

You need to go through the breaking change process http://aka.ms/bcforapi.

**Comment by audunn** [comment]
> > @lirenhe can we get this moving quick? The breaking change is to address a broken functionality for customers/users.
> 
> You need to go through the breaking change process http://aka.ms/bcforapi.

@lirenhe will do,  I am waiting to be granted access to that document

**Comment by audunn** [comment]
Note changes to 2019-10-01, 2019-11-01, 2020-02-01, 2020-03-01 and 2020-05-01 are non function and are to address a issue in  a description that can cause confusing with customers/users and should be corrected

**Comment by audunn** [comment]
@lirenhe Still waiting to be granted access to that document for the breaking change policy. Is there anything that can be done to speed up the process.

**Comment by audunn** [comment]
Removed validation limits on conditional number value that cause issues in generated SDK code.

**Comment by audunn** [comment]
@lirenhe Note the service is in preview status

**Comment by audunn** [comment]
@pilor @lirenhe Can we revisit this I don't believe this is breaking change as this fixes recently added functionality that is broken for SDK users due to issues in generated SDK code and has never worked as expected and should be fixed.

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-06-01/netapp.json [hunk 1]
```diff
@@ -4128,18 +4124,11 @@
...
-          "description": "List of volumes",
-          "type": "array",
-          "items": {
-            "type": "object",
-            "readOnly": true,
-            "description": "Associated volume resource Id"
-          }
-        }
+      "description": "List of volumes",
+      "type": "array",
```
Comment (pilor) [inline specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-06-01/netapp.json:4128]: This is a significant breaking change to what existed before and doesn't meet the RPC defined collection response.  It should be { "value": [{yourObject1}, {yourObject2}] }.

**Review by pilor — CHANGES_REQUESTED** [review]

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-06-01/netapp.json (hunk 1 above)
Comment (audunn) [reply specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-06-01/netapp.json:4128]: Yes, this was to match the actual response from the service as SDK's where having deserialization issues. 
Probably we would need to update the service to produce { "value": [{yourObject1}, {yourObject2}] } type response. Is this a hard requirement and something that what we should do here?

Comment (audunn) [reply specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-06-01/netapp.json]: updated spec

Comment (pilor) [reply specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-06-01/netapp.json]: The collection response is a standard contract that everyone should follow for collection GETs.

**Review by pilor — APPROVED** [review]

**Comment by audunn** [comment]
@lirenhe  this is approved by ARM review can we revisit this breaking change issue? I don't believe this is breaking change as this fixes recently added functionality that is broken for SDK users due to issues in generated SDK code and has never worked as expected and should be fixed.

**Comment by audunn** [comment]
Due to service side changes this will resolved in a later API version.

**Pull request closed without merging**